### PR TITLE
[functionalplus] Remove fplus, add functionalplus to instead of fplus.

### PR DESCRIPTION
**Pull request by LilyWangLL**
Closes https://github.com/microsoft/vcpkg/discussions/38213.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Please fix commit an PR title: It is spelled `fplus`.

File: ports/functionalplus/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    REPO Dobiasd/FunctionalPlus
+    REF "v${VERSION}"
+    SHA512 7b3e451af977ef52a112ad59a4b1f4dc64b120348acb80b751a01bc56d986358b6237d461fe0cf91431b6bc49bc823f8b4ac0016ede9a9e85561cf278d00bb41
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DFunctionalPlus_INSTALL_CMAKEDIR=share/FunctionalPlus
```
Comment (dg0yt) [inline ports/functionalplus/portfile.cmake:12]:
```suggestion
        -DFunctionalPlus_INSTALL_CMAKEDIR=share/functionalplus
```

File: ports/functionalplus/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (dg0yt) [inline ports/functionalplus/portfile.cmake:17]: Header-only. Better add an early
~~~
set(VCPKG_BUILD_TYPE release)  # header-only
~~~
and remove this line.

File: ports/functionalplus/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
```
Comment (dg0yt) [inline ports/functionalplus/portfile.cmake:18]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
```

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/functionalplus/portfile.cmake (hunk 2 above)
Comment (LilyWangLL) [reply ports/functionalplus/portfile.cmake]: Thanks for your suggestions, done.

**Review by Cheney-W — DISMISSED** [review]

**Comment by JavierMatosD** [comment]
Please add a description to the new `functionalPlus` port mentioning that this is the new `fplus` port. That way if a user `vcpkg search fplus` then they will get the `functionalPlus` port.

Also, modify the `frugallyDeep` port to add the new `functionalPlus` port as a dependency. 

Finally, remove `fplus` from the baseline and delete the port directory. Please leave the version files. 

Thanks for the update! :)

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**